### DISCUSSION TITLE
How to contribute to NueJs?

### DISCUSSION THREAD

**Discussion by yaten2302** [nuejs/nue#322] [Q&A]
Hey NeuJs team!!

I'm Yaten, I'm an active open source contributor and want to contribute to NueJs. I've good knowledge of React, JavaScript and TypeScript.
I was checking the roadmap of nuejs on the official website <br>
![image](https://github.com/user-attachments/assets/e5788ba6-12a5-4b2c-b5fc-2ffa422a2e31)

I saw that there are a lot of releases yet to be published. Can anyone please guide that how can we contribute to the same? Like, there are very few open issues, so like, will the new issues will be created or the roadmap is for the core team only?

Thanks

**Comment by tipiirai** [comment]
Hey there and thanks for your interest! Indeed, the bigger future items are not there on the roadmap. They are such epic monsters that it's impossible to predict beforehand what they consist of, and usually the task list is entirely different than planned when the execution proceeds. 

My role could be characterized as [Benevolent dictator for life](https://en.wikipedia.org/wiki/Benevolent_dictator_for_life), and I'm getting a helping hand from the community. @nobkd being the most active and helpful (thanks, man). I'm slowly proceeding with those items on my own pace, right now my hands are dirty with the templates and CSS. 

Currently, there are two massive items where help would be totally awesome. 

1. VS code plugin for the HTML template syntax and Nuemark
2. The client-side, reactive templating language, which has several issues and would need someone to take responsibility (this is pretty challenging project, but I just wanted to ask :)

Do you think you could help with those? Or do you want smaller items?

**Reply by nobkd** [reply]
1. Ref: #319
